packages/state-react/src/lib/useValue.ts

```diff
@@ -49,38 +49,36 @@ export function useValue() {
 	const deps = args.length === 3 ? args[2] : [args[0]]
 	const name = args.length === 3 ? args[0] : `useValue(${args[0].name})`
 
-	const isInRender = useRef(true)
-	isInRender.current = true
+	let isInRender = true
```

Comment:
this is not equivalent to using the useRef, and it won't work as intended because this let binding is scoped to each render individually while a useRef is scoped to the component's lifetime. setting isInRender to true will have no effect on a computed that was created during a previous render.

Reply:
I might be getting tripped up on react lifecycle here. I can drop this back to use the ref.

You've got a unit test for this edge case though, right? It passes on this PR, which is weird if the ref is needed.